src/main/java/core/basesyntax/FileReaderImpl.java

```
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import core.interfaces.FileReader;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+
+    @Override
+    public List<String> read(String path) {

```

Add validation for the `path` parameter and fail fast with an informative exception when it is null or blank. The project requirements explicitly say to handle invalid inputs (for example: "Null parameters"). Consider throwing an IllegalArgumentException like: `if (path == null || path.isBlank()) throw new IllegalArgumentException("Path must not be null or blank");` (this helps tests and callers).